conda/gateways/download.py

```diff
@@ -73,7 +73,9 @@ def download(url, target_full_path, md5sum):
                 with open(target_full_path, 'wb') as fh:
                     streamed_bytes = 0
                     for chunk in resp.iter_content(2 ** 14):
-                        streamed_bytes += len(chunk)
+                        # chunk could be the decompressed form of the real data
+                        # but we want the exact number of bytes read till now
+                        streamed_bytes = resp.raw.tell()
```

Comment:
Does this need to be inside the for loop, or can we just call it once after the loop exits and right before line 89/91

    if content_length and streamed_bytes != content_length?

?

Does my question make sense?